webapp/src/components/modals/channel_settings/channel_settings_internal.jsx

```diff
@@ -113,21 +115,54 @@ export default class ChannelSettingsModalInner extends PureComponent {
     };
 
     handleProjectChange = (id, value) => {
-        const projectKey = value;
+        let projects = value;
+        if (!projects) {
+            projects = [];
+        } else if (!Array.isArray(projects)) {
+            projects = [projects];
+        }
 
-        const filters = {
-            projects: [value],
-            events: [],
-            issue_types: [],
+        let filters = {
+            ...this.state.filters,
+            projects,
         };
+
+        // User has removed a project from selection. Remove any irrelevant selected choices from the events and issue types.
+        if (projects.length < this.state.filters.projects.length) {
+            const issueOptions = getIssueValuesForMultipleProjects(this.props.jiraProjectMetadata, projects);
+            const customFields = getCustomFieldValuesForProjects(this.props.jiraIssueMetadata, projects);
+
+            const selectedIssueTypes = this.state.filters.issue_types.filter((issueType) => {
+                return Boolean(issueOptions.find((it) => it.value === issueType));
+            });
+
+            const selectedEventTypes = this.state.filters.events.filter((eventType) => {
+                if (eventType.includes('customfield')) {
```

Comment:
I forget -- does it always start with 'customfield' or it's always somewhere inside the type name? If it starts with, should use `startsWith` just in case.

Reply:
The event is `event_updated_${fieldId}` where `fieldId` would be something like `customfield_10001` here.